cmd/api/main.go

```diff
@@ -169,7 +170,50 @@ func createBook(w http.ResponseWriter, r *http.Request) {
 
 /* Returns a list of the stored books. */
 func getBooks(w http.ResponseWriter, r *http.Request) {
-	returnedBooks, err := listBooks()
+
+	err := r.ParseForm() //Verify errors in request query params and populates r.Form with its values.
```

Comment:
This should be used for other types of requests (the ones with the `x-www-form-urlencoded` application type header).
Please use `r.URL.Query()`, it the correct one for this case https://pkg.go.dev/net/url#example-URL.Query.

Reply:
https://go.dev/play/p/HOCbBF8c9UI